src/tz.cpp

```diff
@@ -208,7 +209,14 @@ get_known_folder(const GUID& folderid)
     if (SUCCEEDED(hr))
     {
         co_task_mem_ptr folder_ptr(pfolder);
-        folder = std::string(folder_ptr.get(), folder_ptr.get() + wcslen(folder_ptr.get()));
+        const wchar_t* fptr = folder_ptr.get();
+        auto state = std::mbstate_t();
+        const auto required = std::wcsrtombs(nullptr, &fptr, 0, &state);
+        if (required && ~required)
```

Comment:
Why `required && ~required`?

Reply:
Do you mind the suboptimal code generation on msvc? Well, you know, it's the same bad one in both cases when you compile for x86 - even on the most recent compiler version.

I've updated the PR.